[ews-build] Split iOS layout test queue into no-WPT / only-WPT configurations

ryanhaddad opened the pull request:
#### ac46799851df424f061cb3f1ee3475e9a7306586
<pre>
[ews-build] Split iOS layout test queue into no-WPT / only-WPT configurations
<a href="https://bugs.webkit.org/show_bug.cgi?id=254559">https://bugs.webkit.org/show_bug.cgi?id=254559</a>
rdar://107291888

Reviewed by Jonathan Bedard and Aakash Jain.

In order to reduce the turnaround time for iOS layout test results in EWS, split the tests
into no-WPT / only-WPT configurations that can run in parallel.

* Tools/CISupport/ews-app/ews/common/buildbot.py:
(Buildbot):
* Tools/CISupport/ews-app/ews/common/github.py:
(GitHubEWS):
* Tools/CISupport/ews-app/ews/views/statusbubble.py:
* Tools/CISupport/ews-build/config.json:
* Tools/CISupport/ews-build/factories.py:
(iOSTestsOnlyWPTFactory):
(iOSTestsNoWPTFactory):
* Tools/CISupport/ews-build/steps.py:
(RunWebKitTestsInStressMode):
(RunWebKitTestsOnlyWPT):
(RunWebKitTestsOnlyWPT.__init__):
(RunWebKitTestsOnlyWPT.setLayoutTestCommand):
(RunWebKitTestsNoWPT):
(RunWebKitTestsNoWPT.__init__):
(RunWebKitTestsNoWPT.setLayoutTestCommand):
* Tools/CISupport/ews-build/steps_unittest.py:
(TestRunWebKitTestsNoWPT):
(TestRunWebKitTestsNoWPT.setUp):
(TestRunWebKitTestsNoWPT.tearDown):
(TestRunWebKitTestsNoWPT.configureStep):
(TestRunWebKitTestsNoWPT.test_success):

Canonical link: <a href="https://commits.webkit.org/262673@main">https://commits.webkit.org/262673@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/5d8a4fecf486245923e066b04a9a74e9a3dea1b2

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/2118 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/26/builds/2150 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/2216 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/3043 "Built successfully") | [  ~~🛠 wincairo~~](https://ews-build.webkit.org/#/builders/32/builds/2175 "EWS skipped this build as PR had skip-ews label when EWS attempted to process it.") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/2093 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/23/builds/2230 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/2195 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/1930 "EWS skipped this build as PR had skip-ews label when EWS attempted to process it.") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/2137 "Passed tests") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/15/builds/1899 "EWS skipped this build as PR had skip-ews label when EWS attempted to process it.") | [  ~~🧪 api-mac~~](https://ews-build.webkit.org/#/builders/18/builds/1901 "EWS skipped this build as PR had skip-ews label when EWS attempted to process it.") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/2894 "Built successfully") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/13/builds/1884 "EWS skipped this build as PR had skip-ews label when EWS attempted to process it.") | [  ~~🧪 mac-wk1~~](https://ews-build.webkit.org/#/builders/10/builds/1877 "EWS skipped this build as PR had skip-ews label when EWS attempted to process it.") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/1782 "EWS skipped this build as PR had skip-ews label when EWS attempted to process it.") | 
| | [  ~~🛠 tv~~](https://ews-build.webkit.org/#/builders/7/builds/1953 "EWS skipped this build as PR had skip-ews label when EWS attempted to process it.") | [  ~~🧪 mac-wk2~~](https://ews-build.webkit.org/#/builders/36/builds/1925 "EWS skipped this build as PR had skip-ews label when EWS attempted to process it.") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/21/builds/3061 "EWS skipped this build as PR had skip-ews label when EWS attempted to process it.") | 
| | [  ~~🛠 tv-sim~~](https://ews-build.webkit.org/#/builders/30/builds/1947 "EWS skipped this build as PR had skip-ews label when EWS attempted to process it.") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/17/builds/1744 "EWS skipped this build as PR had skip-ews label when EWS attempted to process it.") | | 
| [✅ 🧪 services](https://ews-build.webkit.org/#/builders/28/builds/2100 "Passed tests") | [  ~~🛠 watch~~](https://ews-build.webkit.org/#/builders/4/builds/1874 "EWS skipped this build as PR had skip-ews label when EWS attempted to process it.") | [  ~~🧪 mac-wk2-stress~~](https://ews-build.webkit.org/#/builders/8/builds/1887 "EWS skipped this build as PR had skip-ews label when EWS attempted to process it.") | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/31/builds/2057 "Built successfully") | | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/249 "Built successfully and passed tests") | | | | 
<!--EWS-Status-Bubble-End-->

nt1m commented on Tools/CISupport/ews-build/steps_unittest.py:
```diff
@@ -2231,6 +2231,128 @@ def test_failure(self):
+                                 '--no-build', '--no-show-results', '--no-new-test-results', '--clobber-old-results',
+                                 '--release', '--results-directory', 'layout-test-results', '--debug-rwt-logging',
+                                 '--exit-after-n-failures', '10', '--skip-failing-tests',
+                                 '--exclude-tests', 'imported/w3c/web-playform-tests'],
```
```suggestion
                                 '--exclude-tests', 'imported/w3c/web-platform-tests'],
```

nt1m commented on Tools/CISupport/ews-build/steps_unittest.py:
```diff
@@ -2231,6 +2231,128 @@ def test_failure(self):
+                                 '--no-build', '--no-show-results', '--no-new-test-results', '--clobber-old-results',
+                                 '--release', '--results-directory', 'layout-test-results', '--debug-rwt-logging',
+                                 '--exit-after-n-failures', '10', '--skip-failing-tests',
+                                 '--exclude-tests', 'imported/w3c/web-playform-tests'],
```
```suggestion
                                 '--exclude-tests', 'imported/w3c/web-platform-tests'],
```

nt1m commented on Tools/CISupport/ews-build/steps_unittest.py:
```diff
@@ -2231,6 +2231,128 @@ def test_failure(self):
+                                 '--no-build', '--no-show-results', '--no-new-test-results', '--clobber-old-results',
+                                 '--release', '--results-directory', 'layout-test-results', '--debug-rwt-logging',
+                                 '--exit-after-n-failures', '10', '--skip-failing-tests',
+                                 'imported/w3c/web-playform-tests'],
```
```suggestion
                                 'imported/w3c/web-platform-tests'],
```

nt1m commented on Tools/CISupport/ews-build/steps_unittest.py:
```diff
@@ -2231,6 +2231,128 @@ def test_failure(self):
+                                 '--no-build', '--no-show-results', '--no-new-test-results', '--clobber-old-results',
+                                 '--release', '--results-directory', 'layout-test-results', '--debug-rwt-logging',
+                                 '--exit-after-n-failures', '10', '--skip-failing-tests',
+                                 'imported/w3c/web-playform-tests'],
```
```suggestion
                                 '--exit-after-n-failures', '10', '--skip-failing-tests',
                                 'imported/w3c/web-platform-tests'],
```

ryanhaddad replied:
At least I was consistent with my typo.. Thanks for catching!

robert-jenner commented on Tools/CISupport/ews-build/config.json:
```diff
@@ -175,15 +175,22 @@
       "configuration": "release", "architectures": ["x86_64"],
       "triggered_by": ["ios-16-sim-build-ews"],
-      "workernames": ["ews121", "ews122", "ews123", "ews124", "ews125", "ews126", "ews184", "ews185", "ews200","ews201","ews202","ews203"]
+      "workernames": ["ews121", "ews122", "ews123", "ews124", "ews125", "ews126"]
```
ews121, ews122, ews123, ews124, ews125, ews126, ews184, ews185 

Will continue to serve iOS-16-Simulator-WK2-Tests-EWS 

and we will need to add ews190, which is a new hostname to that queue as well.

robert-jenner commented on Tools/CISupport/ews-build/config.json:
```diff
@@ -175,15 +175,22 @@
+      "factory": "iOSTestsOnlyWPTFactory", "platform": "ios-simulator-16",
+      "configuration": "release", "architectures": ["x86_64"],
+      "triggered_by": ["ios-16-sim-build-ews"],
+      "workernames": ["ews184", "ews185", "ews200","ews201","ews202","ews203"]
```
for the iOS-16-Simulator-WPT-WK2-Tests-EWS we will need to add the following new hostnames to the config file, as well as add them to the WPT queue:

ews191, ews192, ews193, ews194, ews195, ews196, ews197, ews198

ryanhaddad committed:
[ews-build] Split iOS layout test queue into no-WPT / only-WPT configurations
https://bugs.webkit.org/show_bug.cgi?id=254559
rdar://107291888

Reviewed by Jonathan Bedard and Aakash Jain.

In order to reduce the turnaround time for iOS layout test results in EWS, split the tests
into no-WPT / only-WPT configurations that can run in parallel.

* Tools/CISupport/ews-app/ews/common/buildbot.py:
(Buildbot):
* Tools/CISupport/ews-app/ews/common/github.py:
(GitHubEWS):
* Tools/CISupport/ews-app/ews/views/statusbubble.py:
* Tools/CISupport/ews-build/config.json:
* Tools/CISupport/ews-build/factories.py:
(iOSTestsOnlyWPTFactory):
(iOSTestsNoWPTFactory):
* Tools/CISupport/ews-build/steps.py:
(RunWebKitTestsInStressMode):
(RunWebKitTestsOnlyWPT):
(RunWebKitTestsOnlyWPT.__init__):
(RunWebKitTestsOnlyWPT.setLayoutTestCommand):
(RunWebKitTestsNoWPT):
(RunWebKitTestsNoWPT.__init__):
(RunWebKitTestsNoWPT.setLayoutTestCommand):
* Tools/CISupport/ews-build/steps_unittest.py:
(TestRunWebKitTestsNoWPT):
(TestRunWebKitTestsNoWPT.setUp):
(TestRunWebKitTestsNoWPT.tearDown):
(TestRunWebKitTestsNoWPT.configureStep):
(TestRunWebKitTestsNoWPT.test_success):

Canonical link: https://commits.webkit.org/262673@main
